pylint/extensions/kwargs.py

```diff
@@ -0,0 +1,93 @@
+# Licensed under the GPL: https://www.gnu.org/licenses/old-licenses/gpl-2.0.html
+# For details: https://github.com/PyCQA/pylint/blob/main/LICENSE
+# Copyright (c) https://github.com/PyCQA/pylint/blob/main/CONTRIBUTORS.txt
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from astroid import BoundMethod, UnboundMethod, nodes
+
+from pylint.checkers import BaseChecker
+from pylint.checkers.utils import safe_infer
+from pylint.interfaces import INFERENCE
+
+if TYPE_CHECKING:
+    from pylint.lint import PyLinter
+
+
+class KeywordChecker(BaseChecker):
+    """Checks related to keyword arguments."""
+
+    name = "kwargs"
+    msgs = {
+        "W3501": (
+            "Call to `%s` missing keyword argument `%s`.",
+            "consider-using-keyword-argument",
+            "When using a literal directly in a function call, it's very hard to know what the argument is"
+            " if a positional argument is used because there's no variable name or attribute name to "
+            "rely on when reading. Using a keyword argument there's at least the attribute name to "
+            "help the reader understanding the call.",
+        ),
+    }
+
+    def visit_call(self, node: nodes.Call) -> None:
+        """Check that called functions/methods are provided keyword arguments."""
+        called = safe_infer(node.func)
+
+        if not called:
+            return
+
+        needed_keywords, default_kwarg_names = self._get_args(called)
+
+        if len(needed_keywords) - len(default_kwarg_names) < 2:
+            # This checker will not apply if there are less than 2 positional args.
+            return
+
+        provided_kwarg_names = [
+            kwarg.arg for kwarg in node.keywords
+        ] + default_kwarg_names
+
+        for arg_name in needed_keywords:
+            if arg_name not in provided_kwarg_names:
+                self.add_message(
```

Comment:
It would be nice to have one message instead of multiple.

Reply:
I personally disagree here, I think each missing kwarg should be its own msg. Maybe another maintainer can break the tie? @Pierre-Sassoulas

Reply:
I also prefer one message, it's better not to spam and we should also handle the case where 20 kwargs are warning and would make the message too long imo.